plugin-packs/postcss-bundler/src/postcss-import/lib/base64-encoded-import.ts

```diff
@@ -0,0 +1,24 @@
+import { Condition } from './conditions';
+import { formatImportPrelude } from './format-import-prelude';
+
+export function base64EncodedConditionalImport(prelude: string, conditions: Array<Condition>): string {
+	conditions.reverse();
+	const first = conditions.pop()!;
+	let params = `${prelude} ${formatImportPrelude(
+		first.layer,
+		first.media,
+		first.supports,
+	)}`;
+
+	for (const condition of conditions) {
+		params = `'data:text/css;base64,${Buffer.from(
+			`@import ${params}`,
+		).toString('base64')}' ${formatImportPrelude(
+			condition.layer,
+			condition.media,
+			condition.supports,
+		)}`;
+	}
+
+	return params;
+}
```

Comment:
pretty gnarly bit of magic.
Better to have it in its own file.

Reply:
Thanks for the thorough comments here!